MDANSE_GUI/Src/MDANSE_GUI/Tabs/Views/PlotDataView.py

```diff
@@ -72,41 +238,81 @@ def convert_vectors_to_datasets(
     """
     if isinstance(source, h5py.File):
         filename = source.filename
-        parent_dset = source[main_dstet]
+        parent_dset = source[main_dset]
         qvals = parent_dset["q"][:]
         nshells = len(qvals)
-        modq_per_shell = [shell_to_modq(n, parent_dset) for n in range(nshells)]
-        if not all(
-            "custom_field" in parent_dset[f"shell_{shell_index}/qvector_array"].attrs
-            for shell_index in range(nshells)
+        valid_shells = [
+            index
+            for index in range(nshells)
+            if len(parent_dset[f"shell_{index}/qvector_array"][0])
+        ]
+        modq_per_shell = [shell_to_modq(n, parent_dset) for n in valid_shells]
+        available_vectors = np.array([len(qvecs) for qvecs in modq_per_shell])
+        if all(
+            "weights" in parent_dset[f"shell_{shell_index}"]
+            for shell_index in valid_shells
         ):
-            available_vectors = np.array([len(qvecs) for qvecs in modq_per_shell])
-        else:
-            available_vectors = np.array(
+            vec_weights = [
+                parent_dset[f"shell_{shell_index}/weights"][:]
+                for shell_index in valid_shells
+            ]
+            unique_vectors = np.array(
+                [
+                    len(parent_dset[f"shell_{shell_index}/weights"])
+                    for shell_index in valid_shells
+                ]
+            )
+            all_vectors = np.array(
                 [
-                    [
-                        int(x)
-                        for x in parent_dset[
-                            f"shell_{shell_index}/qvector_array"
-                        ].attrs["custom_field"]
-                    ]
-                    for shell_index in range(nshells)
+                    sum(parent_dset[f"shell_{shell_index}/weights"][:])
+                    for shell_index in valid_shells
                 ]
             )
```

Comment:
By defining a generator within this function (so that `valid_shells` and `parent_dset` are captured in the closure):
```python
        def q_data(elem) -> Generator:
            for index in valid_shells:
                yield parent_dset[f"shell_{index}/{elem}"]
```
, a lot of this could be reduced:

```suggestion
        if all("weights" in dat for dat in q_data("")):
            vec_weights = [
                dat[:] for dat in q_data("weights")
            ]
            unique_vectors = np.array(
                [
                    len(dat) for dat in q_data("weights")
                ]
            )
            all_vectors = np.array(
[
                    sum(dat[:]) for dat in q_data("weights")
]
)
```